pkg/declarativeconfig/auth_provider.go

```diff
@@ -0,0 +1,67 @@
+package declarativeconfig
+
+// RequiredAttribute is representation of storage.AuthProvider_RequiredAttribute that supports transformation from YAML.
+type RequiredAttribute struct {
+	AttributeKey   string `yaml:"key,omitempty"`
+	AttributeValue string `yaml:"value,omitempty"`
+}
+
+// ClaimMapping represents a single entry in "claim_mappings" field in auth provider proto.
+type ClaimMapping struct {
+	Path string `yaml:"path,omitempty"`
+	Name string `yaml:"name,omitempty"`
+}
+
+// OIDCConfig contains config values for OIDC auth provider.
+type OIDCConfig struct {
+	Issuer                    string `yaml:"issuer,omitempty"`
+	CallbackMode              string `yaml:"mode,omitempty"`
+	ClientID                  string `yaml:"clientID,omitempty"`
+	ClientSecret              string `yaml:"clientSecret,omitempty"`
+	DisableOfflineAccessScope bool   `yaml:"disableOfflineAccessScope,omitempty"`
+	DontUseClientSecretConfig bool   `yaml:"dontUseClientSecretConfig,omitempty"`
+}
+
+// SAMLConfig contains config values for SAML auth provider.
+type SAMLConfig struct {
+	SpIssuer     string `yaml:"spIssuer,omitempty"`
+	MetadataURL  string `yaml:"metadataURL,omitempty"`
+	CertPEM      string `yaml:"certPEM,omitempty"`
+	SsoURL       string `yaml:"ssoURL,omitempty"`
+	NameidFormat string `yaml:"nameIdFormat,omitempty"`
+}
+
+// IAPConfig contains config values for IAP auth provider.
+type IAPConfig struct {
+	Audience string `yaml:"audience,omitempty"`
+}
+
+// UserpkiConfig contains config values for User Certificates auth provider.
+type UserpkiConfig struct {
+	Keys string `yaml:"keys,omitempty"`
```

Comment:
Those are not keys, but rather CA certificates, so should name this something like CAPems?